### PR TITLE
Fix/5813

### LINKED ISSUE #5813: Column FilterData function example
I'm using the data-filter-data attribute to customize the select filter values of a column through a function but the select values don't work.

I would like an example to use the function on this attribute.

**data-filter-data=func:functionName**

### PR DESCRIPTION
**Pull request by UtechtDustin**
**🤔Type of Request**
- [x] **Bug fix**
- [ ] **New feature**
- [ ] **Improvement**
- [ ] **Documentation**
- [ ] **Other**

**🔗Resolves an issue?**

Fix #5813

**📝Changelog**

- [ ] **Core**
- [x] **Extensions**

Reacreation of the filter control elements, if data was loaded using the `load` method.

**💡Example(s)?**

Before: https://live.bootstrap-table.com/code/UtechtDustin/8508
After: https://live.bootstrap-table.com/code/UtechtDustin/8509

**☑️Self Check before Merge**

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: src/extensions/filter-control/utils.js [hunk 1]
```diff
@@ -353,6 +353,8 @@ export function createControls (that, header) {
     $.each(header.find('th'), (i, th) => {
       const $th = $(th)
 
+      $th.find('.filter-control').remove()
```
Comment (djhvscf) [inline src/extensions/filter-control/utils.js:356]: Is this needed?

**Review by djhvscf — CHANGES_REQUESTED** [review]

File: src/extensions/filter-control/utils.js (hunk 1 above)
Comment (UtechtDustin) [reply src/extensions/filter-control/utils.js:356]: Yes, without that we would append the input/selects again, again and again.
Which means we would have X Input/Elements per Header.

**Comment by djhvscf** [comment]
Thanks @UtechtDustin

File: src/extensions/filter-control/bootstrap-table-filter-control.js [hunk 2]
```diff
@@ -154,6 +154,12 @@ $.BootstrapTable = class extends $.BootstrapTable {
     super.init()
   }
 
+  load (data) {
+    super.load(data)
+
+    UtilsFilterControl.createControls(this, UtilsFilterControl.getControlContainer(this))
```
Comment (djhvscf) [inline src/extensions/filter-control/bootstrap-table-filter-control.js:160]: Should we add an if statement?

```js
if (!this.options.filterControl) {
  return;
}
```

**Review by djhvscf — APPROVED** [review]

File: src/extensions/filter-control/bootstrap-table-filter-control.js (hunk 2 above)
Comment (UtechtDustin) [reply src/extensions/filter-control/bootstrap-table-filter-control.js:164]: yes we should, thanks for the hint.
Added in the last commit.

**Review by djhvscf — APPROVED** [review]

**Pull request merged**